src/core/queryClient.ts

```diff
@@ -282,7 +282,11 @@ export class QueryClient {
     const [filters, options] = parseFilterArgs(arg1, arg2, arg3)
 
     const promises = notifyManager.batch(() =>
-      this.queryCache.findAll(filters).map(query => query.fetch())
+      this.queryCache.findAll(filters).map(query =>
+        query.fetch(undefined, {
+          meta: { refetchPage: options?.refetchPage },
```

Comment:
I played around with this a bit, but I'm not in love with the public api. Two things:

1) if you pass `refetchPage` on a normal useQuery (non-infinite-query), it doesn't do anything
2) if you have a queryKey, but no filters, you'd need to do:

```
queryClient.invalidateQueries('myKey', undefined, {
    refetchPage: (page, allPages) => page === allPages[0]
})
```
yes, you could do:

```
queryClient.invalidateQueries({ queryKey: 'myKey' }, {
    refetchPage: (page, allPages) => page === allPages[0]
})
```

instead, but the same drawback applies when you want to invalidate _everything_:

```
queryClient.invalidateQueries(undefined, {
    refetchPage: (page, allPages) => page === allPages[0]
})
```

---

we could alternatively add that `refetchPage` option to `queryFilters`, similar to `refetchActive` and `refetchInactive` flags we can already provide to `invalidateQueries`. It's not really a query "filter" per se, because it doesn't narrow down anything, but neither is `refetchActive`, so maybe that distinction doesn't matter that much?

---

the only thing that comes to my mind to solve issue 1) would be a separate `invalidateInfiniteQueries`, but that would also mean we'd need `refetchInfiniteQueries` and `resetInfiniteQueries` with all the overloads and I don't think that's worth it.

---

right now, I would go with adding it to the query filters. What do you think @tannerlinsley ?

Reply:
I found a reason to maybe keep the current api: the `refetch` method returned from `useInfiniteQuery` already takes `RefetchOptions`, so it's rather easy to make `refetchPage` work there as well

Reply:
I think I would prefer just adding to the query filters and having it be a noop for non-infinite queries. Thoughts?

Reply:
yep, I can do that, let me see how that looks. Would we still want to have it as an option for `refetch` returned from `useQuery` ? I would say yes, and we could just add it to the options there as well

Reply:
what do you think?

56a3e12b

Reply:
Love it.